src/communication/CommandLine.cpp

```diff
@@ -586,6 +712,55 @@ void CommandLine::loadJSONSettings(const rapidjson::Value& element, Settings& se
     }
 }
 
+std::optional<std::unordered_map<std::string, std::unordered_map<std::string, std::string>>> CommandLine::readResolvedJsonValues(const std::filesystem::path& json_filename)
```

Comment:
Maybe make an alias for this type ?

Reply:
https://github.com/Ultimaker/CuraEngine/pull/2082/commits/549f4a9fa8bf5b42bcf935138962c8319687f7b4